documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -351,4 +353,232 @@ Control characters are not allowed in a specification.
+
+**How to fix the violation**: Ensure that the `x-ms-paths` is overloading an existing url path in the `paths` section.
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="S2001" />S2001 AvoidNestedProperties
+**Output Message**: Consider using x-ms-client-flatten to provide a better end user experience
+
+**Description**: Nested properties can result into bad user experience especially when creating request objects. `x-ms-client-flatten` flattens the model properties so that the users can analyze and set the properties much more easily.
+
+**Why the rule is important**: Overly nested properties (especially required ones) can result into a bad user experience.
+
+**How to fix the violation**: Either eliminate nesting or use the `x-ms-client-flatten` property for a better user experience. More details about the extension can be found [here](https://github.com/Azure/azure-rest-api-specs/blob/dce4da0d748565efd2ab97a43d0683c2979a974a/documentation/swagger-extensions.md#x-ms-client-flatten).
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R3008" />R3008 CollectionObjectPropertiesNaming
+**Output Message**: Collection object '{0}' returned by list operation '{1}' with 'x-ms-pageable' extension, has no property named 'value'. 
+
+**Description**: Per ARM guidelines, a model returned by an `x-ms-pageable` operation must have a property named `value`. This property indicates what type of array the object is.
+
+**Why the rule is important**: To maintain consistency on how `x-ms-pageable` operations and corresponding response objects are modeled and to enable execution of other validation rules based on this consistent structure.
+
+**How to fix the violation**: Ensure that the response object has a property named `value` of `array` type.
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="S2027" />S2027 DefaultMustBeInEnum
+**Output Message**: The default value is not one of the values enumerated as valid for this element.
+
+**Description**: The value assigned as a default for an enum property must be present in the enums' list.
+
+**Why the rule is important**: SDKs generated may fail to compile if we try to enforce a default value that is not a part of the enums defined in the list.
```

Comment:
How about "Generated SDKs in types languages may fail to compile if we try to enforce a default value that is not a part of the enums defined in the list and in other languages may fail in serialization/deserialization phases."